kill.md: change signal explanation (closes #616)

waldyrious opened the pull request:
I've rebased, improved the descriptions, and fixed linting issues in #616. This had to be done in a separate PR for the reasons explained in #996, but the original commits by @lanzhiwang remain properly attributed.

/cc @igorshubovych, @markusthoemmes, @rubenvereecken, @ostera.

agnivade commented:
Love this page. I always forget the signal codes. LGTM !

jedahan commented:
Really great contribution

ostera commented on pages/common/kill.md:
```diff
@@ -1,12 +1,28 @@
+
+- Terminate a program using the SIGHUP (hang up) signal. Many daemons will reload instead of terminating:
+
+`kill {{1|-HUP}} {{process_id}}`
```
Shouldn't this be `-1`? And if so, wouldn't it be more concise to have `-{{1|HUP}}` instead?

waldyrious replied:
Good catch! I'll make the changes :)

ostera commented:
Good stuff guys, keep it up 👍

waldyrious commented:
@ostera I personally would have liked this one merged rather than squashed, since the commits were logically separated and properly described -- per the description above:

> I've rebased, improved the descriptions, and fixed linting issues in #616. 

 I'll assume it was just an oversight, but if you actually disagree, let me know :)

ostera commented:
I figured that since the other commits were unattributed it wouldn't be much of an issue – I agree that if we have multiple authors on the same PR, we shouldn't squash to preserve authorship on the individual changes.

waldyrious commented:
It's not as much about the authorship (although that counts too, to encourage participation). I was more concerned about encouraging (and preserving) the proper separation of commits as atomic changes, with informative descriptions.
